### PR TITLE
Issue 1744 - Zombie Process accumulation issue

### PR DESCRIPTION
**Pull request by saramsey**
This should fix the zombie process issue. Would be good to review the code changes.  Current theory is that the zombie processes were accumulating because the installed SIGCHLD handler would reap only one child process. So in cases where there are two child processes exiting at nearly the same time, my guess is the handler got called only once, resulting in a "leak" of a zombie child process. In any event, only way to be sure is to deploy to `arax.ncats.io` and see if the leak stops. I have tested this code on `arax-backup.rtx.ai` using the ARAX browser UI, the `test_ARAX_api.py` pytest script, and the `test_ARAX_api_failure_modes.py` pytest script, and things look good. No zombies:
```
rt@5d5ebba937d1:/mnt/data/orangeboard/kg2/RTX/code/ARAX/test_production_api$ ps axwf
  PID TTY      STAT   TIME COMMAND
  257 pts/1    Ss     0:00 bash
  905 pts/1    S      0:00  \_ su - rt
  906 pts/1    S      0:00      \_ -su
  982 pts/1    R+     0:00          \_ ps axwf
    1 pts/0    Ss+    0:00 bash
  709 ?        Ss     0:00 /usr/sbin/apache2 -k start
  712 ?        Sl     0:00  \_ /usr/sbin/apache2 -k start
  713 ?        Sl     0:00  \_ /usr/sbin/apache2 -k start
  854 ?        S      0:28 python3 -u -m openapi_server
  887 ?        Sl     0:06 python3 -u -m openapi_server
  953 ?        R      2:03  \_ python3 -u -m openapi_server
```

### CODE REVIEW

**Comment by saramsey** [comment]
This PR gives the SIGCHLD handler the ability to reap as many child processes as are available to reap, rather than just exiting after the first child process is reaped. I would imagine the only time there is more than one child process available to reap would be under load, but that is conjecture.

**Review by finnagin — APPROVED** [review]

**Comment by saramsey** [comment]
@edeutsch finn and I reviewed the code and think it looks OK. We are waiting for your take, before we merge. Just let us know if we should go ahead and merge to `master`.

**Review by edeutsch — APPROVED** [review]

**Comment by edeutsch** [comment]
I have not tested it, but it looks reasonable.

**Comment by saramsey** [comment]
OK, merged to `master`. How should we roll it out?

**Comment by edeutsch** [comment]
I can roll it out if you like

**Pull request merged**
